### PR TITLE
impl Default for remaining point types

### LINKED ISSUE #154: Point types should `impl Default`
We can't write a generic `impl<T: Identity> Default for T` because of the orphan rules, but we should provide `Default` impl for every type implementing `Identity`.

### PR DESCRIPTION
**Pull request by isislovecruft**
Closes #154.

### CODE REVIEW

File: src/montgomery.rs [hunk 1]
```diff
@@ -155,6 +155,12 @@ impl Identity for ProjectivePoint {
     }
 }
 
+impl Default for ProjectivePoint {
+    fn default() -> ProjectivePoint {
+        ProjectivePoint::identity()
+    }
+}
+
```
Comment (hdevalence) [inline src/montgomery.rs]: Should we add this? We don't expose the `ProjectivePoint` externally, and we don't use the `Default` impl internally, so this function is dead code.

**Comment by isislovecruft** [comment]
I'll remove it.

**Comment by isislovecruft** [comment]
Should I also remove the impls for `ProjectiveNielsPoint` and `AffineNielsPoint`?

**Comment by hdevalence** [comment]
Nope, they're used when building lookup tables. I don't have a problem with having a `Default` impl for an internal point type, it's just that since it's not used by the `ProjectivePoint` I would rather not add it unless we needed it for something.

**Pull request merged**